apps/tangle-dapp/utils/convertRewardPointsToReward.ts

```diff
@@ -0,0 +1,37 @@
+import { BN_QUINTILL } from '@polkadot/util';
+
+const BIG_INT_QUINTILL = BigInt(BN_QUINTILL.toString());
+
+/**
+ * Convert the reward points to reward in token
+ * Following the `do_payout_stakers` logic in the tangle
+ * to convert the reward points to rewards
+ * see https://github.com/webb-tools/tangle/blob/8c1be851059d21fe524ca30808219d5b26c01713/pallets/roles/src/functions.rs#L508-L539
+ * @param rewardPoints the reward points (got from `api.query.roles.erasRestakeRewardPoints`)
+ * @param validatorCommision the validator commission (got from `api.query.staking.erasValidatorPrefs`)
+ * @param ownStaked the own staked amount of the validator (got from `api.query.staking.erasStakersOverview`)
+ * @param totalStaked the total staked amount of the validator (got from `api.query.staking.erasStakersOverview`)
+ * @returns the reward in token
+ */
+export default function convertRewardPointsToReward(
```

Comment:
I think it's best if we can output `BN` here and then pass it down to the charts to handle

Reply:
I converted it to a bigint since the reward points are of number type. However, after the calculation, we need to scale up to Tangle units, which consist of 18 decimals. I opted for bigint here because the worker only supports JavaScript primitive types.